app/lib/publisher/models.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart' as db;
+
+/// Canonical publisher data.
+@db.Kind(name: 'Publisher', idType: db.IdType.String)
+class Publisher extends db.ExpandoModel {
+  /// The associated domain name of the publisher.
+  String get domainName => id as String;
+
+  /// Markdown formatted description of the publisher.
+  ///
+  /// Limited to 64 KB in length.
+  @db.StringProperty()
+  String description;
+
+  @db.StringProperty()
+  String websiteUrl;
+
+  @db.DateTimeProperty()
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime updated;
+}
+
+/// Derived publisher data.
+@db.Kind(name: 'PublisherInfo', idType: db.IdType.String)
+class PublisherInfo extends db.ExpandoModel {
+  /// The associated domain name of the publisher.
+  String get domainName => id as String;
+
+  @db.DateTimeProperty()
+  DateTime updated;
+
+  /// List of packages that are associated with this publisher.
+  @db.StringListProperty()
+  List<String> packages;
+
+  /// List of userIds that are administrators of this publisher.
+  /// (their e-mail address is public information)
+  @db.StringListProperty()
+  List<String> admins;
+
+  /// List of userIds that are public members of this publisher.
+  /// (their e-mail address is public information)
+  @db.StringListProperty()
+  List<String> publicMembers;
+}
+
+/// Values for [PublisherMember.role]
+abstract class PublisherMemberRole {
```

Comment:
didn't we drop roles from the initial design?

Reply:
I think, just having `member` and `pending` would be good.

Reply:
How would you track the administrators of a publisher?

Reply:
I think we just say that all members of a publisher are equal... all are administrators..

Reply:
But we need to display at least one admin e-mail for the publisher. Some of the motivation behind publishers was that some team members don't want to expose their e-mail address while being able to publish the package.

Reply:
Oh, I was thinking we would have a contact email... and have a validation flow for said email address.

I imagine that users would often want to use a mailinglist... which isn't a specific user, but an email that can be validated..

Reply:
Does Google's oauth flow work against an e-mail list?

Reply:
Doubtful, an email list isn't a Google account... you could perhaps create a google account for your email list, but this seems unnatural...

Reply:
I've added `contactEmail` to the `Publisher` entity to reflect this. However, this also means we should only have `admin` and `pending`, as first everybody that is being added to the publisher will be full-access administrator. Maybe I just leave it there for now, we can easily check if the consts is in use or not.